infrastructure/simulation-tool/test/operations/changePubKey.test.ts

```diff
@@ -0,0 +1,219 @@
+            ];
+
+            await executePubKeyChanges(preparedPubKeyChanges, 0);
+
+            expect(senderWallet.setSigningKey).to.have.been.calledThrice;
+        });
+
+        it('should call executePubKeyChange with given delay between calls', async () => {
+            const senderWallet = sinon.createStubInstance(RollupWallet);
+            const preparedPubKeyChanges: PreparedPubKeyChange[] = [
+                { from: senderWallet, ethAuthType: 'ECDSA', feeToken: constants.AddressZero },
+                { from: senderWallet, ethAuthType: 'ECDSA', feeToken: constants.AddressZero },
+                { from: senderWallet, ethAuthType: 'ECDSA', feeToken: constants.AddressZero }
+            ];
+            const delay = 100;
+            const expectedTotalTime = delay * (preparedPubKeyChanges.length - 1);
+
+            const startTime = Date.now();
+            await executePubKeyChanges(preparedPubKeyChanges, delay);
+            const endTime = Date.now();
+
+            const actualTotalTime = endTime - startTime;
+            expect(actualTotalTime).to.be.greaterThan(expectedTotalTime);
+        });
+
+        it('should return array of promises', async () => {
+            const senderWallet = sinon.createStubInstance(RollupWallet);
+            const preparedPubKeyChanges: PreparedPubKeyChange[] = [
+                { from: senderWallet, ethAuthType: 'ECDSA', feeToken: constants.AddressZero },
+                { from: senderWallet, ethAuthType: 'ECDSA', feeToken: constants.AddressZero },
+                { from: senderWallet, ethAuthType: 'ECDSA', feeToken: constants.AddressZero }
+            ];
+
+            const actualResults = await executePubKeyChanges(preparedPubKeyChanges, 0);
+
+            const allResultsAreTransactions = actualResults.reduce((acc, next) => {
+                acc = acc && next instanceof Promise;
+                return acc;
+            }, true);
+
+            expect(allResultsAreTransactions).to.be.true;
+        });
+
+        it('should return array of promised transactions', async () => {
+            const senderWallet = sinon.createStubInstance(RollupWallet);
+            senderWallet.setSigningKey.resolves(sinon.createStubInstance(Transaction));
+            const preparedPubKeyChanges: PreparedPubKeyChange[] = [
+                { from: senderWallet, ethAuthType: 'ECDSA', feeToken: constants.AddressZero },
+                { from: senderWallet, ethAuthType: 'ECDSA', feeToken: constants.AddressZero },
+                { from: senderWallet, ethAuthType: 'ECDSA', feeToken: constants.AddressZero }
+            ];
+
+            const actualResults = await Promise.all(
+                (await executePubKeyChanges(preparedPubKeyChanges, 0)).map(async (tx) => await tx)
+            );
+
+            const allResultsAreTransactions = actualResults.reduce((acc, next) => {
+                acc = acc && next instanceof Transaction;
+                return acc;
+            }, true);
```

Comment:
Unless there is a specific reason, I'd use [Array.every](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/every)

Reply:
done